tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -635,4 +651,30 @@ function() {
 		$tag = wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) );
 		$this->assertNotSame( $tag, webp_uploads_img_tag_update_mime_type( $tag, 'the_content', $attachment_id ) );
 	}
+
+	/**
+	 * Tests whether additional mime types generated only for allowed image sizes or not.
+	 *
+	 * @test
+	 */
+	public function it_should_create_mime_types_for_allowed_sizes_only() {
+		add_filter(
+			'webp_uploads_image_sizes_with_additional_mime_type_support',
+			function( $sizes ) {
+				$sizes['allowed_size_400x300'] = 'allowed_size_400x300';
+				return $sizes;
+			}
+		);
+
+		add_image_size( 'allowed_size_400x300', 400, 300, true );
+		add_image_size( 'not_allowed_size_200x150', 200, 150, true );
```

Comment:
I think for full coverage, we should add another test that does the same thing, but without the filter and instead manually setting that new field in the `$_wp_additional_image_sizes` global (which will later become available as a `add_image_size` parameter.

Reply:
Good idea. Added one more test.